### ISSUE TITLE
prefixLengthRange needs more than integer type for prefix lists

### ISSUE THREAD

**Issue by JeffGiroux** [F5Networks/f5-declarative-onboarding#237]
### Environment
 * Declarative Onboarding Version: 1.21
 * BIG-IP Version: 15 and 16

### Summary

The routing commands expect an integer for prefixLengthRange when creating a prefix list. However, this is not an integer and should be intepreted as range. Maybe I'm doing the wrong syntax. I've tried a few ways.

### Steps To Reproduce
1. Submit the following declaration:
```json
{
    "schemaVersion": "1.6.0",
    "class": "Device",
    "Common": {
    	"class": "Tenant",
    	"hostname": "example.local"
    }
}
```

2. Observe the following error response:
```json
{
    "id": "62b57a65-ad1a-4ac4-8ddb-0427fad81b79",
    "selfLink": "https://localhost/mgmt/shared/declarative-onboarding/task/62b57a65-ad1a-4ac4-8ddb-0427fad81b79",
    "code": 400,
    "status": "ERROR",
    "message": "bad declaration",
    "errors": [
        "Unable to parse request body. Should be JSON format."
    ],
    "result": {
        "class": "Result",
        "code": 400,
        "status": "ERROR",
        "message": "bad declaration",
        "errors": [
            "Unable to parse request body. Should be JSON format."
        ]
    },
    "declaration": {}
}
```

### Expected Behavior
This class object should be able to support what TMSH can support. See below for possible values and range values.

```
modify net routing prefix-list myPrefixList1 entries modify { 10 { prefix-len-range 
Values:
  Enter a valid prefix range. Example for IPv4:
    1:32                       ge 1 and le 32
    1:                         ge 1
    :32                        le 32
    32                         ge 32 le 32
  For IPv6:
    1:128                      ge 1 and le 128
```

I should be able to have a block like this and DO interpret it correctly...
```json
        "myPrefixList1": {
            "class": "RoutingPrefixList",
            "entries": [
                {
                    "name": 10,
                    "action": "permit",
                    "prefix": "10.0.0.0/8",
                    "prefixLengthRange": 16:
                }
            ]
        },
```

Or maybe instead of type integer it is type string and you can take a block like this with quotes around "16:"...
```json
        "myPrefixList1": {
            "class": "RoutingPrefixList",
            "entries": [
                {
                    "name": 10,
                    "action": "permit",
                    "prefix": "10.0.0.0/8",
                    "prefixLengthRange": "16:"
                }
            ]
        },
```

### Actual Behavior

Originally I tried 16 as schema says type integer only according, but it results in ge 16 le 16. I need simply "ge 16" which is equivalent to 16:

Then I tried 16:, but DO fails...
```
error: Unexpected token : in JSON at position 2415
```

Then I tried putting quotes around "16:" but DO fails...
```
2021-05-21T01:38:57.376Z [29340]: info: {"id":"c8dad054-1da9-4351-9646-c91c995f7f19","selfLink":"https://localhost/mgmt/shared/declarative-onboarding/task/c8dad054-1da9-4351-9646-c91c995f7f19","code":400,"status":"ERROR","message":"bad declaration","errors":[{"keyword":"type","dataPath":".declaration.Common['myPrefixList1'].entries[0].prefixLengthRange","schemaPath":"#/allOf/12/then/properties/entries/items/properties/prefixLengthRange/type","params":{"type":"integer"},"message":"should be integer"}],"result":{"class":"Result","code":400,"status":"ERROR","message":"bad declaration","errors":[{"keyword":"type","dataPath":".declaration.Common['myPrefixList1'].entries[0].prefixLengthRange","schemaPath":"#/allOf/12/then/properties/entries/items/properties/prefixLengthRange/type","params":{"type":"integer"},"message":"should be integer"}]},"declaration":{"schemaVersion":"1.20.0","class":"Device","async":true,"label":"Onboard BIG-IP","Common":{"class":"Tenant","hostname":"giroux2-f7ae-f5vm01.example.com","myDns":{"class":"DNS","nameServers":["169.254.169.253"],"search":["f5.com"]},"myNtp":{"class":"NTP","servers":["169.254.169.123"],"timezone":"UTC"},"azureuser":{"class":"User","userType":"regular","shell":"bash","keys":["ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC0M0mYLo14yuXbKuL+CKiZ/et+gt8grphon65I+NrKNjiLMaeV43vVOKfnMrYD+BRxzEVp6guqmIBAkvUq4UZm1Xc9nWwRve/0mqVDf1Z9HYgUx5nYSJ6qQD5Bnl+UjS8gUQoP2tD7HhPBuJZHDRTa9ZluHiBazpnmbfatjlXbv4go5OfzOZsKOdPl4CBi971yCITb82nvo7+VOgMwvcI07O8ct13w/6UdSss2KTxmso7lZrRMX0ax9YVbgqhff2kB+Q8umi3XOVGUUCtx0ioRUFIBchl5pSKuWntMrknVNiLTsTmQ7SiS819d4cnSnXzHqBPt+pe7evuqFY8hbbnv giroux@FLD-ML-00026870 "],"partitionAccess":{"all-partitions":{"role":"admin"}}},"external":{"class":"VLAN","tag":4093,"mtu":1500,"interfaces":[{"name":"1.1","tagged":false}],"cmpHash":"dst-ip","failsafeEnabled":false,"failsafeAction":"failover-restart-tm","failsafeTimeout":90},"external-self":{"class":"SelfIp","address":"10.255.10.4/24","vlan":"external","allowService":["tcp:179"],"trafficGroup":"traffic-group-local-only"},"default":{"class":"Route","gw":"10.255.10.1","network":"default","mtu":1500,"localOnly":false},"internal":{"class":"VLAN","tag":4094,"mtu":1500,"interfaces":[{"name":"1.2","tagged":false}],"cmpHash":"dst-ip","failsafeEnabled":false,"failsafeAction":"failover-restart-tm","failsafeTimeout":90},"internal-self":{"class":"SelfIp","address":"10.255.20.4/24","vlan":"internal","allowService":"default","trafficGroup":"traffic-group-local-only"},"myPrefixList1":{"class":"RoutingPrefixList","entries":[{"name":10,"action":"permit","prefix":"10.0.0.0/8","prefixLengthRange":"16:"}]},"myRouteMap1":{"class":"RouteMap","entries":[{"name":"10","action":"permit","match":{"ipv4":{"address":{"prefixList":"myPrefixList1"}}}}]},"myBGP":{"class":"RoutingBGP","localAS":65530,"addressFamilies":[{"internetProtocol":"ipv4","redistributionList":[{"routingProtocol":"kernel","routeMap":"myRouteMap1"}]}],"gracefulRestart":{"gracefulResetEnabled":true,"restartTime":120},"neighbors":[{"address":"10.255.255.4","peerGroup":"Neighbor"},{"address":"10.255.255.5","peerGroup":"Neighbor"}],"peerGroups":[{"name":"Neighbor","addressFamilies":[{"internetProtocol":"ipv4","softReconfigurationInboundEnabled":true}],"remoteAS":65515}]}}}}
```

## Full declaration
Here's my DO file
```json
{
    "schemaVersion": "1.20.0",
    "class": "Device",
    "async": true,
    "label": "Onboard BIG-IP",
    "Common": {
        "class": "Tenant",
        "hostname": "{{{ HOST_NAME }}}.example.com",
        "myDns": {
            "class": "DNS",
            "nameServers": [
                "169.254.169.253"
            ],
            "search": [
                "f5.com"
            ]
        },
        "myNtp": {
            "class": "NTP",
            "servers": [
                "169.254.169.123"
            ],
            "timezone": "UTC"
        },
        "{{{ USER_NAME }}}": {
            "class": "User",
            "userType": "regular",
            "password": "{{{ ADMIN_PASS }}}",
            "shell": "bash",
            "keys": ["{{{ SSH_KEYS }}}"],
            "partitionAccess": {
                "all-partitions": {
                    "role": "admin"
                }
            }
        },
        "external": {
            "class": "VLAN",
            "tag": 4093,
            "mtu": 1500,
            "interfaces": [
              {
                "name": "1.1",
                "tagged": false
              }
            ],
            "cmpHash": "dst-ip"
        },
        "external-self": {
            "class": "SelfIp",
            "address": "10.255.10.4/24",
            "vlan": "external",
            "allowService": [
                "tcp:179"
            ],
            "trafficGroup": "traffic-group-local-only"
        },
        "default": {
            "class": "Route",
            "gw": "10.255.10.1",
            "network": "default",
            "mtu": 1500
        },
        "internal": {
            "class": "VLAN",
            "tag": 4094,
            "mtu": 1500,
            "interfaces": [
              {
                "name": "1.2",
                "tagged": false
              }
            ],
            "cmpHash": "dst-ip"
        },
        "internal-self": {
            "class": "SelfIp",
            "address": "10.255.20.4/24",
            "vlan": "internal",
            "allowService": "default",
            "trafficGroup": "traffic-group-local-only"
        },
        "myPrefixList1": {
            "class": "RoutingPrefixList",
            "entries": [
                {
                    "name": 10,
                    "action": "permit",
                    "prefix": "10.0.0.0/8",
                    "prefixLengthRange": "16:"
                }
            ]
        },
        "myRouteMap1": {
            "class": "RouteMap",
            "entries": [
                {
                    "name": "10",
                    "action": "permit",
                    "match": {
                        "ipv4": {
                            "address": {
                                "prefixList": "myPrefixList1"
                            }
                        }
                    }
                }
            ]
        },
        "myBGP": {
            "class": "RoutingBGP",
            "localAS": 65530,
            "addressFamilies": [
                {
                    "internetProtocol": "ipv4",
                    "redistributionList": [
                        {
                            "routingProtocol": "kernel",
                            "routeMap": "myRouteMap1"
                        }
                    ]
                }
            ],
            "gracefulRestart": {
                "gracefulResetEnabled": true,
                "restartTime": 120
            },
            "neighbors": [
                {
                    "address": "10.255.255.4",
                    "peerGroup": "Neighbor"
                },
                {
                    "address": "10.255.255.5",
                    "peerGroup": "Neighbor"
                }
            ],
            "peerGroups": [
                {
                    "name": "Neighbor",
                    "addressFamilies": [
                        {
                            "internetProtocol": "ipv4",
                            "softReconfigurationInboundEnabled": true
                        }
                    ],
                    "remoteAS": 65515
                }
            ]
        }
    }
}
```

**Comment by dstokesf5** [comment]
Thank you for reporting this. I agree that it looks like this property needs to be a string. I have added this issue to our internal product backlog as AUTOTOOL-2524.

**Comment by dstokesf5** [comment]
This issue has been resolved with the 1.23.0 release of Declarative Onboarding.
